### PR TITLE
feat: Implement cluster health check endpoints

### PR DESCRIPTION
**Pull request by purplefox**
### Description 

This is a stacked PR. Please only review the top commit(s).

This PR implements the endpoints /chc/live and /chc/ready. They were previously implemented using Jetty resource extensions, but this will not work when Jetty disappears.

These endpoints simply return an empty JsonObject. They differ from other endpoints in that they need to be available when the server is waiting for preconditions.

### Testing done 

Includes tests cases for the endpoints.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java [hunk 1]
```diff
@@ -116,6 +118,9 @@ int actualPort() {
   private Router setupRouter() {
     final Router router = Router.router(vertx);
 
+    router.route(HttpMethod.GET, "/chc/ready").handler(ServerVerticle::chcHandler);
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java:121]: Can we add a comment explaining the importance of handling these before the server state handler (and auth handler)? It's great that you've added tests for this but I think a comment would be nice for anyone who happens to be glancing at the code.

Comment (purplefox) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java:121]: >> Are we confident that users don't rely on this functionality?

I did a search in all confluentinc repos and /chc seems to be the only user of this. In terms of community users this was never a documented feature, afaik.

Comment (vcrfxia) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java:123]: OK. Let's prepare a blurb for the changelog/release notes with a summary of implications from the migration from Jetty to Vert.x, including that some Jetty-specific plugins/features may not be supported in full? There's almost certainly a more artful way to word this but I think it'd be good to call out.

The auto-generated changelog will pick it up if the commit message (the one that actually gets merged to GitHub) includes `BREAKING CHANGE: <text here>`. Or we can manually insert it into the changelog/release notes come release time.

Comment (purplefox) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java:123]: Hmm, AIUI the fact that we were previously using Jetty was an implementation detail, and we've never had any documented features reliant on it.

**Pull request merged**